Jetpack Cloud: Tracks event alterations and additions for launch

rcoll opened the pull request:
#### Changes proposed in this Pull Request

* Alters all existing tracks events to use the `jetpack_cloud_` source and context
* Adds several other tracks events using the proper source and context

#### Testing instructions

* Load up all screens within Jetpack Cloud. Check for console errors or warnings.
* Feel free to run through the entire testing process for tracks events.

Fixes #

enejb commented:
Why did we change the events away from calypso? (What is the rational there) Do they still work as expected? 

I think keeping it as calypso was intentional. 

1. They would work as expected. Since they currently as I understand it need to be prefixed with calypso. (Only certain prefixes are allowed) 
2. The code lives in calypso repo it makes it easier to find when someone is looking for the event. 
3. It would mean that if we ever use the same section in calypso or vise versa the event names would not be so strange. :)

rcoll commented:
All feedback addressed, changed event name prefixes all to `calypso_jetpack_` and then either a `scan`, `backup`, or `sidebar` context, and finally, a description. Also addressed the missed events @rcanepa :-) Thanks all.
